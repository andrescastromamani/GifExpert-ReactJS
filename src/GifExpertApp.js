import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import PropTypes from 'prop-types';

export const GifExpertApp = ({ categoriesGifs = [] }) => {
    const [categories, setcategories] = useState(categoriesGifs);
    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory setCategories={setcategories} />
            <hr></hr>

            <ol>
                {
                    categories.map(category =>
                    (<GifGrid
                        key={category}
                        category={category} />
                    ))
                }
            </ol>
        </>
    )
}
GifExpertApp.propTypes = {
    categoriesGifs: PropTypes.array.isRequired
}
