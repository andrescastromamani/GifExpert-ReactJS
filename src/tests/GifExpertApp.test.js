import { shallow } from "enzyme"
import { GifExpertApp } from "../GifExpertApp";

describe('GifExpertApp', function () {
    test('should be a function', function () {
        const categoriesGifs = [
            'music', 'reggae'
        ]
        const wrapper = shallow(<GifExpertApp categoriesGifs={categoriesGifs} />);
        expect(wrapper).toMatchSnapshot();
    })
    test('should show a list of categories', function () {
        const categoriesGifs = [
            'music', 'reggae'
        ]
        const wrapper = shallow(<GifExpertApp categoriesGifs={categoriesGifs} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categoriesGifs.length);
    })
})