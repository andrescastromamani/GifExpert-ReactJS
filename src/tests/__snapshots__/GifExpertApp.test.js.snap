// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should be a function 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="music"
      key="music"
    />
    <GifGrid
      category="reggae"
      key="reggae"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp should show a list of categories 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="music"
      key="music"
    />
    <GifGrid
      category="reggae"
      key="reggae"
    />
  </ol>
</Fragment>
`;
